frontend/cypress/README.md

```diff
@@ -11,11 +11,12 @@ Installed all dev dependencies from frontend folder. Ensure the `baseUrl` field
 Before you start using Cypress suite, you might need export some environment variables - depending on environment where tests are executed. If your authentication method defaults to `anonymous` **(i.e. dev env), no actions are needed.**
 
 ```bash
-export CYPRESS_BASE_URL=<value>        # defaults to http://localhost:3000
-export CYPRESS_USERNAME=<value>        # defaults to jenkins, opt. kubeadmin
-export CYPRESS_PASSWD=<value>          # no defaults 
-export CYPRESS_AUTH_PROVIDER=<value>   # defaults to my_htpasswd_provider, 
-                                       # or optionally openshift for AWS
+export CYPRESS_BASE_URL=<value>           # defaults to http://localhost:3000
+export CYPRESS_USERNAME=<value>           # defaults to jenkins, opt. kubeadmin
+export CYPRESS_PASSWD=<value>             # no defaults
+export CYPRESS_AUTH_PROVIDER=<value>      # defaults to my_htpasswd_provider,
+export CYPRESS_AUTH_PROVIDER_NAME=<value> # defaults to my_htpasswd_provider. Can pass a custom name for the htp provider.
```

Comment:
See my previous comment. This name doesn't match the code.

Reply:
> I can't get this to work:
> 
> ```
> make -e CYPRESS_AUTH_PROVIDER=htpasswd -e CYPRESS_AUTH_HTTP_PROVIDER_NAME=htpasswd -e CYPRESS_PASSWD=kiali -e CYPRESS_USERNAME=kiali -e CYPRESS_BASE_URL=https://kiali-istio-system.apps.kialici.ABC.com cypress-gui
> ```

It should now be:
```
make -e CYPRESS_BASE_URL="https://kiali-istio-system.apps-crc.testing" -e CYPRESS_PASSWD="PW" CYPRESS_USERNAME="kubeadmin" CYPRESS_AUTH_PROVIDER="anyauth" CYPRESS_AUTH_HTP_PROVIDER_NAME="anyauth" cypress-gui
```
I tested this on my CRC cluster and it worked fine.

Reply:
> Also, I'm confused. Why are both "my_htpasswd_provider"? Shouldn't it be:
> `CYPRESS_AUTH_PROVIDER=(openshift|aws|htpasswd)`
> and if it is htpasswd THEN you use CYPRESS_AUTH_PROVIDER_NAME which defaults to my_htpasswd_provider but can be any custom name you happen to use for your htpasswd provider???

Yeah... thinking about this more it might make more sense just to have a boolean here and not even worry about the actual name. `CYPRESS_HTP_PROVIDER=true` or something. Because the name is non standard for the provider it _can_ be anything, right?

Reply:
can we fetch this provider name somwhere from API?

Reply:
You can have multiple providers in OpenShift. So you could look at the OAuth object at the identityProviders, but there could in theory be more than one (and that is why we see buttons in the login form - you need to tell OpenShift which provider you are authenticating with). In this case, we need to tell the Cypress stuff which provider needs to authenticate the credentials we are passing in.

Reply:
Now it just uses whatever you passed in to `CYPRESS_AUTH_PROVIDER` and assumes it follows the `htpasswd` flow. Works even when multiple providers are present.